server/configuration.go

```diff
@@ -27,6 +46,22 @@ func (c *configuration) Clone() *configuration {
 	return &clone
 }
 
+func (c *configuration) IsValid() error {
+	if len(c.ProvisioningServerURL) == 0 {
+		return errors.New("must specify ProvisioningServerURL")
+	}
+	_, err := url.Parse(c.ProvisioningServerURL)
+	if err != nil {
+		return errors.Wrap(err, "invalid ProvisioningServerURL")
+	}
+
+	if len(c.InstallationDNS) == 0 {
+		return fmt.Errorf("must specify InstallationDNS")
+	}
+
+	return err
+}
+
 // getConfiguration retrieves the active configuration under lock, making it safe to use
 // concurrently. The active configuration may change underneath the client of this method, but
 // the struct returned by this API call is considered immutable.
```

Comment:
Copying comment from Lev:

nit: 0/5 return a copy of configuration and avoid the struct mutability issue altogether.